saja/task_manager.py

```diff
@@ -0,0 +1,199 @@
+                        task = UrgentTask(
+                            title=item['title'],
+                            description=item['description'],
+                            due_date=item['due_date'],
+                            status=status,
+                            priority=priority
+                        )
+                        self.urgent_tasks.append(task)
+                    else:
+                        task = Task(
+                            title=item['title'],
+                            description=item['description'],
+                            due_date=item['due_date'],
+                            status=status
+                        )
+                        self.tasks.append(task)
+        except FileNotFoundError:
+            print(f"File {self.data_file} not found. Starting with an empty task list.")
+        except json.JSONDecodeError:
+            print("Error decoding JSON data. Starting with an empty task list.")
+
+    def save_tasks(self):
+        """Save the current tasks back to the JSON data file."""
+        data = []
+        for task in self.tasks + self.urgent_tasks:
+            task_data = {
+                'title': task.title,
+                'description': task.description,
+                'due_date': task.due_date.strftime('%Y-%m-%d'),
+                'status': task.status.name
+            }
+            if isinstance(task, UrgentTask):
+                task_data['priority'] = task.priority.name
+            data.append(task_data)
+
+        with open(self.data_file, 'w') as f:
+            json.dump(data, f, indent=4)
+
+    def add_task(self):
+        """Prompt the user to add a new task, either regular or urgent."""
+        title = get_valid_input("Enter the task title: ")
+        description = get_valid_input("Enter the task description: ")
+        due_date = get_valid_input("Enter the due date (YYYY-MM-DD): ")
+        status_input = get_valid_input("Enter the status (InProgress/Completed): ")
+        status = Status[status_input]
+
+        is_urgent_input = get_valid_input("Is this an urgent task? (yes/no): ").lower()
+        if is_urgent_input in ['yes', 'y']:
+            priority_input = get_valid_input("Enter the priority (Low/Medium/High): ").capitalize()
+            priority = Priority[priority_input]
+            task = UrgentTask(title, description, due_date, status, priority)
+            self.urgent_tasks.append(task)
+        else:
+            task = Task(title, description, due_date, status)
+            self.tasks.append(task)
+
+        self.save_tasks()
+        print("Task added successfully.")
+
+    def view_tasks(self):
+        """Display all regular and urgent tasks."""
+        print("\nRegular Tasks:")
+        for task in self.tasks:
+            task.display()
+
+        print("\nUrgent Tasks:")
+        for task in self.urgent_tasks:
+            task.display()
+
+    def update_task(self):
```

Comment:
maybe we need to put this code under try/catch block ?
because the validation can raise an exceptions.

Reply:
@sajalahalih have you managed to update this ?